### ISSUE TITLE
The player notification remains after calling stop on iOS

### ISSUE THREAD

**Issue by mt633** [ryanheise/just_audio#823]
**Which API doesn't behave as documented, and how does it misbehave?**
Using `just_audio_background`, the iOS stop method is not properly called in `audio_service`, causing the player notification to remain. This works as expected on Android.

**Minimal reproduction project**
The `just_audio_background` example, just changing https://github.com/ryanheise/just_audio/blob/9a67394849e30bacdd2c7d135bbdff0427835430/just_audio_background/example/lib/main.dart#L348 to `player.stop`.

(Note that the example in `audio_service` does work as expected.)

**To Reproduce**
1. Start playback
2. Stop playback
3. Open the lock screen or the Notification Center
4. The notification is still there

**Error messages**
No error message. See https://github.com/ryanheise/just_audio/issues/747#issuecomment-1250924390 for log output.

**Expected behavior**
The notification should disappear

**Smartphone:**
 - Device: iPhone SE (simulator)
 - OS: iOS 15.5

**Flutter SDK version**
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.3.1, on macOS 12.5.1 21G83 darwin-arm)
[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2020.3)
[✓] VS Code (version 1.71.0)
[✓] Connected device (4 available)
[✓] HTTP Host Availability

• No issues found!
```

**Additional context**
The stop code seems to run when a new item is loaded. Not sure if this is connected in any way.

**Comment by mt633** [comment]
After some further digging, I can see that the `processingState` is never set to `AudioProcessingState.idle` with `just_audio_background` whereas it is set in the `audio_service` example. That means that this code in `audio_service` (line 1009-1012) is never run.

```dart
      if (playbackState.processingState == AudioProcessingState.idle &&
          previousState?.processingState != AudioProcessingState.idle) {
        await AudioService._stop();
      }
```

Not sure if that gives you any clues as to what the issue might be, otherwise I'll continue to investigate this further.

**Comment by ryanheise** [comment]
That is interesting, the code responsible for triggering that is the following method in just_audio_background:

```dart
  Future<void> stop() async {
    if (_justAudioEvent.processingState == ProcessingStateMessage.idle) {
      return;
    }
    _updatePosition();
    _playing = false;
    _broadcastState();
    // TODO: We should really stop listening to events here to mimic
    // just_audio's behaviour. E.g. if stop() was called, we actually want to
    // keep the state around even though the platform may be disposing its own
    // state.
    _platform.disposePlayer(DisposePlayerRequest(id: (await _player).id));
    _justAudioEvent = _justAudioEvent.copyWith(
      processingState: ProcessingStateMessage.idle,
    );
  }
```

That would suggest that that first "if" condition is true.

**Comment by mt633** [comment]
I believe the problem here is that `_broadcastState()` is called before setting the state in this line:
```dart
 _justAudioEvent = _justAudioEvent.copyWith(
      processingState: ProcessingStateMessage.idle,
    );
```
Moving the `_broadcastState` method below it resolves the issue for me.

**Comment by mt633** [comment]
Would you accept a PR with that change?

**Comment by ryanheise** [comment]
Ah yes. I think the intention was to broadcast the state with immediate effect before it makes the async call, but then do another broadcast of the state after making the async call. But that is on the assumption that the user should receive immediate feedback. Can you tell me after making that change if you notice any lag between clicking stop and seeing the notification/etc disappear?

**Comment by ryanheise** [comment]
Yes, PRs are welcome. I'll leave that method unmodified in case you would like to contribute a PR, but in this case you'll need to make your PR based on the `fix/unimplemented_methods` branch.

**Comment by mt633** [comment]
No, as far as I can tell there is no lag between clicking stop and seeing the notification disappear.

**Comment by devyueyue** [comment]
Hello, has the issue of clicking to stop playing and the notification on the lock screen page not disappearing been resolved?

I also have this problem.
